### PR TITLE
reorder modifications

### PR DESCRIPTION
**Pull request by JacquesBorsenberger**
Signed-off-by: jacques borsenberger <jacques.borsenberger@rte-france.com>

### CODE REVIEW

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 1]
```diff
@@ -486,6 +486,18 @@ public ResponseEntity<Mono<Void>> applyGroovyScript(@PathVariable("studyUuid") U
         return ResponseEntity.ok().body(studyService.assertCanModifyNode(nodeUuid).then(studyService.assertComputationNotRunning(nodeUuid)).then(studyService.applyGroovyScript(studyUuid, groovyScript, nodeUuid).then()));
     }
 
+    @PutMapping(value = "/studies/{studyUuid}/nodes/{nodeUuid}/network-modification/{modificationId}")
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/study/server/StudyController.java:489]: We don't want something more explicit for the api route ?
For example @PutMapping(value = "/studies/{studyUuid}/nodes/{nodeUuid}/network-modification/reorder/{modificationId}")

Comment (etiennehomer) [reply src/main/java/org/gridsuite/study/server/StudyController.java:489]: Or maybe the before parameter is required and its value is null when we want to set the modification to the end ?

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 2]
```diff
@@ -486,6 +487,18 @@ public ResponseEntity<Mono<Void>> applyGroovyScript(@PathVariable("studyUuid") U
...
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/study/server/StudyController.java:490]: `modificationId` -> `modificationUuid`

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 3]
```diff
@@ -486,6 +487,18 @@ public ResponseEntity<Mono<Void>> applyGroovyScript(@PathVariable("studyUuid") U
...
+    @Operation(summary = "move network modification before another")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The modification order is updated")})
+    public ResponseEntity<Mono<Void>> moveModification(@PathVariable("studyUuid") UUID studyUuid,
+                                                        @PathVariable("nodeUuid") UUID nodeUuid,
+                                                        @PathVariable("modificationId") UUID modificationUuid,
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/study/server/StudyController.java:495]: `modificationId` -> `modificationUuid`

File: src/main/java/org/gridsuite/study/server/StudyService.java [hunk 4]
```diff
@@ -1825,5 +1825,19 @@ public Mono<Void> deleteNode(UUID studyUuid, UUID nodeId, boolean deleteChildren
             })
             .doOnError(throwable -> LOGGER.error(throwable.toString(), throwable));
     }
+
+    public Mono<Void> reorderModification(UUID studyUuid, UUID nodeUuid, UUID modificationUuid, UUID before) {
+        checkStudyContainsNode(studyUuid, nodeUuid);
+        return networkModificationTreeService.getModificationGroupUuid(nodeUuid).flatMap(groupId ->
+            networkModificationService.reorderModification(groupId, modificationUuid, before)
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/study/server/StudyService.java:1832]: `groupId` -> `groupUuid`

File: src/test/java/org/gridsuite/study/server/StudyTest.java [hunk 5]
```diff
@@ -2268,6 +2268,44 @@ public void testCreateVoltageLevel() {
...
+            .expectStatus().isNotFound();
+
+        webTestClient.put()
+            .uri("/v1/studies/{studyUuid}/nodes/{nodeUuid}/network-modification/{modificationID}?before={modificationID2}",
+                UUID.randomUUID(), modelNodeUuid, modification1, modification2)
+            .exchange()
+            .expectStatus().isForbidden();
+
+        // update switch on first modification node
+        webTestClient.put()
```
Comment (sBouzols) [inline src/test/java/org/gridsuite/study/server/StudyTest.java:2296]: Maybe add a test without `before` parameter

**Review by sBouzols — COMMENTED** [review]
some comments will test after front review

**Review by sBouzols — APPROVED** [review]
code review OK

**Pull request merged**